src/foremast/pipeline/construct_pipeline_block.py

```diff
@@ -153,6 +155,11 @@ def construct_pipeline_block(env='',
         'scalingpolicy': scalingpolicy,
     })
 
+    if data['app']['canary']:
```

Comment:
So as we discussed, I think moving everything under this one conditional is best. It keeps all the canary logic in one place